src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -2896,6 +2927,39 @@ private void AddToFieldsToFetch(string field, string alias)
             FieldsToFetch.Add(new FieldToFetch(field, alias));
         }
 
+        private bool NeedToAddFromAliasToField(string field)
+        {
+            return _addedDefaultAlias &&
+                   field.StartsWith($"{_fromAlias}.") == false &&
+                   field.StartsWith("id(") == false &&
+                   field.StartsWith("counter(") == false &&
+                   field.StartsWith("counterRaw(") == false;
+        }
+
+        private void AddFromAliasToFieldToFetch(ref string field, ref string alias, bool quote = false)
+        {
+            if (field == alias)
+            {
+                alias = null;
+            }
+
+            field = quote
+                ? $"{_fromAlias}.'{field}'"
```

Comment:
I wonder if it works when we have a query like:

```
session.Query<User>()
                        .Select(user => new
                        {
                            user.Group.Order
                        });
```

I'm pretty sure it will work but let's add such test case for that.

Reply:
no actually that didn't work, good thinking.
fixed this + added test case